Initial commits

jsf9k opened the pull request:
## 🗣 Description ##

This pull request includes the initial functionality for this repository.

## 💭 Motivation and context ##

A repo must _do_ something in order to be useful!

Resolves #2.

## 🧪 Testing ##

The testing here is admittedly not great, but the code has been used to:
- Manually generate YML from the Markdown in [cisagov/log4j-affected-db](cisagov/log4j-affected-db).
- Manually generate Markdown from the YML generated in the previous item.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] _All_ future TODOs are captured in issues, which are referenced in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Bump to version 1.0.0.

## ✅ Post-merge checklist ##

- [x] Create a release.

felddy committed:
Add first version of conversion script

- rename a bunch of example identifiers
- remove uneeded files

felddy committed:
Fix bug which caused first table to be skipped

felddy committed:
Add pyyaml requirement to setup.py

felddy committed:
Remove noop assignment

felddy committed:
Add dateparser requirement

felddy committed:
Add convert_cisagov script

felddy committed:
Add pytz requirement

felddy committed:
Move common regular expresion to module level

felddy committed:
Correct docstring

felddy committed:
Fix parser to handle changes in markdown

felddy committed:
Add logic to handle malformed table data

jsf9k committed:
Add some missing types libraries

Also remove types-setuptools, since it isn't being used.

jsf9k committed:
Add code to munge, normalize, and sort YAML outputs

jsf9k committed:
Add a console script for the normalizer

felddy committed:
Trim whitespace from values

felddy committed:
Merge branch 'first-commits' of github.com:cisagov/md-table-to-yml into first-commits

felddy committed:
Add derived "investigated" field from status field

jsf9k committed:
Correctly dump the combined, normalized, and sorted data as YML

jsf9k committed:
Add code to generate the Markdown table from the grand mal YML

jsf9k committed:
Fix logic that checks for empty strings

Also duplicate that logic in a few other places where it is useful.

felddy committed:
Fix crash on unparsable "dates".

Coo coo cachoo!

felddy committed:
Add status field calculation

felddy committed:
Replace pytablewriter with a manual table printer

pytablewriter was creating very wide tables.
there was not an easy way to make more condensed tables

felddy committed:
Remove duplicate requirements

I did not read the "note" that told me not to
do the thang that I did.

jsf9k committed:
Remove blurb specific to skeleton repos

jsf9k committed:
Remove blurb specific to skeleton repos

jsf9k committed:
Update description from skeleton boilerplate

jsf9k committed:
Update description from skeleton boilerplate

jsf9k committed:
Add a section describing the common YML format

jsf9k committed:
Add a section describing the common YML format

felddy committed:
Add missing comma in status value set

Thanks to @jsf9k for spotting this over my virtual shoulder.
Co-authored-by: @jsf9k

felddy committed:
Merge branch 'first-commits' of github.com:cisagov/md-table-to-yml into first-commits

jsf9k committed:
Update the YML description to match the latest proposed format

The new format can support information for multiple log4j-related
CVEs.

jsf9k committed:
Update the YML description to match the latest proposed format

The new format can support information for multiple log4j-related
CVEs.

jsf9k committed:
Add links for each log4j CVE

jsf9k committed:
Add links for each log4j CVE

dav3r committed:
Initial commit of md_from_template code

dav3r committed:
Initial commit of md_from_template code

dav3r committed:
Add chevron requirement and console script for md_from_template

dav3r committed:
Add chevron requirement and console script for md_from_template

dav3r committed:
Merge branch 'first-commits' of github.com:cisagov/md-table-to-yml into first-commits

felddy committed:
Add missing comma in status value set

Thanks to @jsf9k for spotting this over my virtual shoulder.
Co-authored-by: @jsf9k

felddy committed:
Require Python 3.10

The parser is making use of PEP 634:
Structural Pattern Matching: Specification
https://www.python.org/dev/peps/pep-0634/

PEP 634 was released in Python 3.10

There is an outstanding PR with mypy support:
https://github.com/python/mypy/pull/10191

felddy committed:
Merge branch 'first-commits' of github.com:cisagov/md-table-to-yml into first-commits

felddy committed:
Merge branch 'first-commits' of github.com:cisagov/md-table-to-yml into first-commits

felddy committed:
Add support for tracking multiple CVEs

dav3r committed:
Simplify logic by making it less generalized

We just want to get this working - we can make it nicer later.

mcdonnnj committed:
Update the Python version for the lint GHA job

This Python package requires Python >=3.10, so the `lint` job must
install an appropriate Python version to mirror that need.

mcdonnnj committed:
Remove tests that are leftover from the skeleton

These tests do not apply to this project and so they should be removed.

mcdonnnj committed:
Remove package data items from setup

We do not leverage package data in this project.

mcdonnnj committed:
Update console script names

The standard is to use hyphens to separate words.

mcdonnnj committed:
Update the mdyml package to pass tests

mcdonnnj committed:
Fix version imports for packages

mcdonnnj committed:
Update command names in docopt entries for scripts

mcdonnnj committed:
Update to pass bandit checks

mcdonnnj committed:
Disable the mypy pre-commit hook

The current version of mypy does not support the `match` statement that
was added in Python 3.10. Please see python/mypy#10191 to check the
status of support.

mcdonnnj committed:
Update the table detection in convert-cisagov

Update the method of table detection to rely on the number of columns
matching an expected value instead of relying on a particular piece of
text to be in the line.

felddy committed:
Fix detection table header

The table format changed again.

mcdonnnj committed:
Disable the log level test for mdyml

The convert_cisagov module has to run to completion for this test which
is an inordinate amount of time. Disabling for now because that module
will only serve historical purposes in the long term.

mcdonnnj committed:
Update Python version required in GHA job

When updating versions to Python 3.10 the `coveralls-finish` job was
missed. This has been rectified.

mcdonnnj committed:
Add basic testing for md_from_template package

This mirrors the configuration of the tests for the mdyml package.

mcdonnnj committed:
Add basic testing for yml package

This mirrors the configuration of the tests for the mdyml package.

mcdonnnj committed:
Add basic testing for ymlmd package

This mirrors the configuration of the tests for the mdyml package.

dav3r committed:
Document the existence of the md_from_template utility

dav3r committed:
Add links to each module in the README

mcdonnnj committed:
Update the URL for the cisagov software list

mcdonnnj committed:
Use ruamel.yaml for cisagov conversion

The ruamel.yaml library gives the necessary tools to correctly format
outputted YAML in all regards except line length. This will allow us to
pass linting with the generated file with minimal rule modifications.

jsf9k committed:
Modify calculate_status() to analyze data for the default CVE only

Also change references to the "patched_versions" key to
"fixed_versions", in keeping with the generated YML.

jsf9k committed:
Wrangle the multiple vendor links into a single string

jsf9k committed:
Fix indenting error

This was causing the last_updated column to be missing in the case
where the last_updated field already included timezone information.

mcdonnnj committed:
Add minor error handling to yml2md

Handle `KeyError`s when trying to output rows from a read YAML file.

mcdonnnj committed:
Adjust Markdown link extraction regex

Make the regex greedier for the link portion, but also switch to only
matching non-whitespace characters instead of any character.

mcdonnnj committed:
Strip trailing whitespace from row data file

Change md_from_template to strip any trailing whitespace from the end
of the source table data file. This prevents it from injected undesired
whitespace into the rendered template.

mcdonnnj committed:
Fix delimiter row in yml2md

The delimiter row was not updated to match the header row and so the
column lengths did not match.

mcdonnnj committed:
Improve header and delimiter row output logic

Store the header fields in a list and use that as a basis to generate
the appropriate header and delimiter rows in yml2md.

jsf9k committed:
Remove automatically generated fields from the YML description

Also remove the non-default CVEs from the example YML since we are not
supporting those right now.

jsf9k committed:
Rename repo

The new name better reflects the repo contents.

jsf9k commented:
> This pull request **introduces 1 alert** and **fixes 1** when merging [2546e74](https://github.com/cisagov/log4j-md-yml/commit/2546e7408a7d4e0dd32ded030dea82716b93c1ad) into [9697906](https://github.com/cisagov/log4j-md-yml/commit/9697906b8d4f86a2bea8858752b08472ebf663b6) - [view on LGTM.com](https://lgtm.com/projects/g/cisagov/log4j-md-yml/rev/pr-0f4c07bea802340417a48920c2b2ee34cc3e842f)
> 
> **new alerts:**
> 
> * 1 for Syntax error
> 
> **fixed alerts:**
> 
> * 1 for Clear-text logging of sensitive information

The syntax error is due to the use of [the `match` statement](https://docs.python.org/3/whatsnew/3.10.html#pep-634-structural-pattern-matching) added in Python 3.10.  LGTM's scanning has not yet caught up to this late-breaking Python addition.  We can ignore this one.

jsf9k committed:
Use isort from PyCQA/isort#1875

This PR resolves PyCQA/isort#1874, which is currently biting us when
we attempt to install the isort pre-commit hook.

dav3r commented on .coveragerc:
```diff
@@ -2,7 +2,7 @@
 
 [run]
-source = src/example
+source = src/mdyml
```
We have other code directories now:
```suggestion
source =
    src/md_from_template
    src/mdyml
    src/yml
    src/ymlmd
```

jsf9k committed:
Add a TODO comment reminding us to revert the isort change

jsf9k committed:
Add all source directories to pycov configuration

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

dav3r commented on setup.py:
```diff
@@ -1,5 +1,5 @@
 """
-This is the setup module for the example project.
+This is the setup module for the mdyml project.
```
```suggestion
This is the setup module for the log4j-md-yml project.
```

dav3r commented on setup.py:
```diff
@@ -42,19 +42,19 @@ def get_version(version_file):
-    version=get_version("src/example/_version.py"),
-    description="Example Python library",
+    version=get_version("src/mdyml/_version.py"),
+    description="mdyml Python library",
```
```suggestion
    description="log4j-md-yml Python library",
```

dav3r commented on bump_version.sh:
```diff
@@ -6,7 +6,7 @@ set -o nounset
 set -o pipefail
 
-VERSION_FILE=src/example/_version.py
+VERSION_FILE=src/mdyml/_version.py
```
How are we handling the versions of each module in this library since we have a `_version.py` file in each `src/` subdirectory?

jsf9k replied:
We should really just use one version, or split this into three different repos.  I vote for the former.

dav3r replied:
I'm on board with using one version for the entire library also.

jsf9k replied:
Please see commit 41445a6.

dav3r replied:
Works for me, thanks!

dav3r commented on src/mdyml/convert_cisagov.py:
```diff
@@ -0,0 +1,211 @@
+"""convert_cisagov markdown conversion tool.
+
+Fetches the contents of the cisagov GitHub repository and converts it to YAML.
```
Specify repo name in docstring:
```suggestion
Fetches the contents of the cisagov/log4j-affected-db GitHub repository
and converts it to YAML.
```

dav3r commented on src/mdyml/convert_cisagov.py:
```diff
@@ -0,0 +1,211 @@
+
+
+def convert() -> None:
+    """Parse the markdown at the given URL and convert it to YAML."""
+    # Get the markdown
+    logging.info("Reading markdown at %s", RAW_URL)
```
Proper capitalization:
```suggestion
    """Parse the Markdown at the given URL and convert it to YAML."""
    # Get the Markdown
    logging.info("Reading Markdown at %s", RAW_URL)
```

dav3r commented on src/mdyml/convert_cisagov.py:
```diff
@@ -0,0 +1,211 @@
+                case _:  # anything else; unknown, or missing
+                    out_dict["cves"][DEFAULT_CVE_ID]["affected_versions"] = versions
+
+        # Extract link from markdown
```
More capitalization:
```suggestion
        # Extract link from Markdown
```

dav3r commented on src/mdyml/convert_cisagov.py:
```diff
@@ -0,0 +1,211 @@
+                timespec="seconds"
+            )
+
+        # # Sort record keys
+        # in_row_dict = {key: in_row_dict[key] for key in ORDERED_FIELD_NAMES}
```
Can we delete this?
```suggestion
```

dav3r commented on src/mdyml/convert_cisagov.py:
```diff
@@ -0,0 +1,211 @@
+
+
+def main() -> None:
+    """Set up logging and call the mdyml function."""
```
```suggestion
    """Set up logging and call the convert function."""
```

dav3r commented on src/mdyml/convert_cisagov.py:
```diff
@@ -0,0 +1,211 @@
+"""convert_cisagov markdown conversion tool.
```
Proper capitalization:
```suggestion
"""convert_cisagov Markdown conversion tool.
```

dav3r commented on src/mdyml/convert_ncsc_nl.py:
```diff
@@ -0,0 +1,165 @@
+"""convert_ncsc_nl markdown conversion tool.
```
Proper capitalization:
```suggestion
"""convert_ncsc_nl Markdown conversion tool.
```

dav3r commented on src/mdyml/convert_ncsc_nl.py:
```diff
@@ -0,0 +1,165 @@
+"""convert_ncsc_nl markdown conversion tool.
+
+Fetches the contents of the NCSC NL GitHub repository and converts it to YAML.
```
Specify repo name in docstring:
```suggestion
Fetches the contents of the NCSC-NL/log4shell GitHub repository
and converts it to YAML.
```

dav3r commented on src/mdyml/convert_ncsc_nl.py:
```diff
@@ -0,0 +1,165 @@
+
+
+def convert() -> None:
+    """Parse the markdown at the given URL and convert it to YAML."""
```
Capitalization:
```suggestion
    """Parse the Markdown at the given URL and convert it to YAML."""
```

dav3r commented on src/mdyml/convert_ncsc_nl.py:
```diff
@@ -0,0 +1,165 @@
+        row_dict["last_updated"] = datetime.now(timezone.utc).isoformat(
+            timespec="seconds"
+        )
+        # Extract link from markdown
```
Capitalization:
```suggestion
        # Extract link from Markdown
```

dav3r commented on src/mdyml/convert_ncsc_nl.py:
```diff
@@ -0,0 +1,165 @@
+
+
+def main() -> int:
+    """Set up logging and call the mdyml function."""
```
```suggestion
    """Set up logging and call the convert function."""
```

dav3r commented on src/ymlmd/yml2md.py:
```diff
@@ -0,0 +1,162 @@
+
+
+def generate_markdown(software: Software) -> None:
+    """Generate markdown manually."""
```
```suggestion
    """Generate Markdown manually."""
```

dav3r commented on src/ymlmd/yml2md.py:
```diff
@@ -0,0 +1,162 @@
+                        [x for x in default_cve["fixed_versions"] if len(x) != 0]
+                    ),
+                    status=s["status"],
+                    # convert vendor links to markdown links if they start
```
```suggestion
                    # convert vendor links to Markdown links if they start
```

dav3r reviewed (changes requested):
I noted a bunch of minor thangs and one question about how we are handling our module vs. project versions.

jsf9k committed:
Correct repo name

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Correct capitalization

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Specify source repos

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Delete unused code

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Correct function names in comments

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Correct capitalization

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Modify all modules to leverage the same version

The version used is that in the mdyml module.

jsf9k committed:
Modify all modules to leverage the same version

The version used is that in the mdyml package.

dav3r reviewed (approved):
Outstanding! 🧞

mcdonnnj committed:
Revert "Use isort from PyCQA/isort#1875"

This reverts commit 51752eefd97b539db3338554db59dda7354f46b7.

The workaround added in that commit is no longer required, since the
issue was fixed in setuptools 60.1.1.

mcdonnnj commented on README.md:
```diff
@@ -1,25 +1,75 @@
+   [cisagov/log4j-affected-db](https://github.com/cisagov/log4j-affected-db)
+   into one grand YML file: [`normalize-yml`](src/yml/normalize_yml.py)
+1. Generate a Markdown table from the YML output of the previous step:
+   [`yml2md`](src/ymlmd/yml2md.py)
```
Fixing the format name and adding a note to reflect that this functionality does not currently work.

```suggestion
1. (Work in Progress) Translate the Markdown table of vulnerable software
   from [NCSC-NL/log4shell](https://github.com/NCSC-NL/log4shell) into a
   common YAML format: [`convert-nscs-nl`](src/mdyml/convert_ncsc_nl.py)
1. Merge the YAML from the previous step with the YAML from
   [cisagov/log4j-affected-db](https://github.com/cisagov/log4j-affected-db)
   into one grand YAML file: [`normalize-yml`](src/yml/normalize_yml.py)
1. Generate a Markdown table from the YAML output of the previous step:
   [`yml2md`](src/ymlmd/yml2md.py)
```

mcdonnnj commented on README.md:
```diff
@@ -1,25 +1,75 @@
+
+## Common YML format ##
+
+The common YML format looks like this:
```
```suggestion
## Common YAML format ##

The common YAML format looks like this:
```

mcdonnnj commented on README.md:
Adding a key that I believe should be included.

```suggestion
      - https://vendor1.com/discussion/comment/622612/#Comment_622612
    last_updated: "2021-12-09"
  ⋮
```

mcdonnnj commented on README.md:
```suggestion
| `last_updated` | The date of the last update for this product in ISO 8601 timestamp format. |
| `notes` | A free-form text field for additional notes. |
```

mcdonnnj commented on bump_version.sh:
```diff
@@ -6,7 +6,7 @@ set -o nounset
 set -o pipefail
 
-VERSION_FILE=src/example/_version.py
+VERSION_FILE=src/mdyml/_version.py
```
If we want to use one centralized version then I would like to use a method similar to [black](https://github.com/psf/black) where there is a separate module to control versioning as seen:

- In the `setup.py` [here](https://github.com/psf/black/blob/092959ff1f9253347b01eeb2d6d72e15bad7e25a/setup.py#L75-L78) and [here](https://github.com/psf/black/blob/092959ff1f9253347b01eeb2d6d72e15bad7e25a/setup.py#L88) (I know we aren't using SCM controlled versioning. I just wanted to point out the file location).
- In the `black` module [here](https://github.com/psf/black/blob/092959ff1f9253347b01eeb2d6d72e15bad7e25a/src/black/__init__.py#L70).

If we're in agreement I can do the work.

dav3r replied:
I have no problem with doing it that way, but I don't want that to hold us up from getting this MVP out the door.  Can you please make an issue and add a TODO comment for this?

mcdonnnj replied:
Since I'd already poked around with it, this was implemented in 5e53e83.

mcdonnnj commented on src/md_from_template/md_from_template.py:
```diff
@@ -0,0 +1,101 @@
+    print(chevron.render(template, data))
+
+
+def main() -> int:
```
```suggestion
def main() -> None:
```

mcdonnnj commented on src/md_from_template/md_from_template.py:
```diff
@@ -0,0 +1,101 @@
+
+    # Stop logging and clean up
+    logging.shutdown()
+    return 0
```
```suggestion
```

mcdonnnj commented on src/mdyml/__init__.py:
I don't like regex without explanatory comments.

```suggestion
# Match a Markdown link in the form
# [link text](https://example.com/path/to/object)
# with the named match groups text and link.
MD_LINK_RE = re.compile(r"\[(?P<text>.*?)\]\((?P<link>\S*)\)")
```

mcdonnnj commented on src/mdyml/convert_cisagov.py:
```diff
@@ -0,0 +1,209 @@
+        )
+        if len(in_row_list) != EXPECTED_COLUMN_COUNT:
+            logging.warning(
+                "Skipping line %d with unexpected number of columns %d: %s",
```
This number does not correspond to a line in the source file.

```suggestion
                "Skipping row %d with unexpected number of columns %d: %s",
```

mcdonnnj commented on src/mdyml/convert_ncsc_nl.py:
```diff
@@ -0,0 +1,166 @@
+    )
+
+
+def main() -> int:
```
```suggestion
def main() -> None:
```

mcdonnnj commented on src/mdyml/convert_ncsc_nl.py:
```diff
@@ -0,0 +1,166 @@
+
+    # Stop logging and clean up
+    logging.shutdown()
+    return 0
```
```suggestion
```

mcdonnnj commented on src/yml/normalize_yml.py:
```diff
@@ -0,0 +1,94 @@
+    return software
+
+
+def main() -> int:
```
```suggestion
def main() -> None:
```

mcdonnnj commented on src/yml/normalize_yml.py:
```diff
@@ -0,0 +1,94 @@
+
+    # Stop logging and clean up
+    logging.shutdown()
+    return 0
```
```suggestion
```

mcdonnnj commented on src/ymlmd/yml2md.py:
```diff
@@ -0,0 +1,162 @@
+            print(f"{i}: {err} - {s}", file=sys.stderr)
+
+
+def main() -> int:
```
```suggestion
def main() -> None:
```

mcdonnnj commented on src/ymlmd/yml2md.py:
```diff
@@ -0,0 +1,162 @@
+
+    # Stop logging and clean up
+    logging.shutdown()
+    return 0
```
```suggestion
```

mcdonnnj commented on src/ymlmd/yml2md.py:
```diff
@@ -0,0 +1,162 @@
+    for row_contents in [header_fields, ["-" * len(field) for field in header_fields]]:
+        print(f'| {" | ".join(row_contents)} |')
+
+    # Print table converting lists to comma separated strings
```
This doesn't seem correct.

```suggestion
    # Print table converting lists to Markdown table rows
```

mcdonnnj commented on src/ymlmd/yml2md.py:
```diff
@@ -0,0 +1,162 @@
+                    # with http
+                    vendor_links=", ".join(
+                        [
+                            f"[link]({link})" if link.startswith("http") else link
```
I would kind of like to massage URLs to HTTPS, but what does everyone think?

dav3r replied:
So far, we are not really modifying user-entered data for our MVP.  Please make an issue about this and maybe someone can get it done later.

mcdonnnj commented on src/mdyml/convert_cisagov.py:
```diff
@@ -0,0 +1,209 @@
+        row_count += 1
+        in_row_list = row.split("|")[1:-1]
+        logging.debug(
+            "Processing line %d with %d columns: %s", row_count, len(in_row_list), row
```
This number does not correspond to a line in the source file.

```suggestion
            "Processing row %d with %d columns: %s", row_count, len(in_row_list), row
```

mcdonnnj commented on setup.py:
```diff
@@ -74,20 +74,23 @@ def get_version(version_file):
-    python_requires=">=3.6",
+    python_requires=">=3.10",
     # What does your project relate to?
     keywords="skeleton",
```
I'm not sure what else would better fit here. I just know `skeleton` does not.

```suggestion
    keywords="log4j",
```

mcdonnnj reviewed (changes requested):
First pass review with some suggestions and questions.

dav3r committed:
Prefer YAML to YML and add a "Work in Progress" note to one script

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

dav3r committed:
Apply suggestions from code review

Change return value from Int to None

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

dav3r committed:
Prefer YAML to YML

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

dav3r committed:
Use more accurate verbiage in log messages

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

dav3r committed:
Add a helpful explanatory comment for a regex

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

dav3r committed:
Replace skeleton keyword with something useful

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

dav3r committed:
Change return value from Int to None

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

dav3r committed:
Add "last_updated" key and description to YAML documentation

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

dav3r committed:
Correct an outdated comment

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

dav3r committed:
Merge branch 'first-commits' of github.com:cisagov/log4j-md-yml into first-commits

dav3r committed:
Add two additional keywords to setup.py

mcdonnnj committed:
Use a library level module to control versioning

Since we want to use the same version for all packages in this project
it makes sense to define this at the root level instead of pulling in
the version from one of the library's packages.

mcdonnnj commented on README.md:
```diff
@@ -1,25 +1,77 @@
+
+This repository contains Python code to:
+
+1. (Work in Progress) Translate the Markdown table of vulnerable software
```
Since they now have each letter as a separate Markdown file.

```suggestion
1. (Work in Progress) Translate the Markdown tables of vulnerable software
```

mcdonnnj commented on src/mdyml/__init__.py:
```diff
@@ -0,0 +1,26 @@
+import re
+
+# cisagov Libraries
+from _log4j_md_yml_version import __version__  # noqa: F401
```
```suggestion
# Standard Python Libraries
import re

# We disable a Flake8 check for "Module imported but unused (F401)" here because
# although this import is not directly used, it populates the value
# package_name.__version__, which is used to get version information about this
# Python package.
# cisagov Libraries
from _log4j_md_yml_version import __version__  # noqa: F401
```

mcdonnnj commented on src/mdyml/convert_cisagov.py:
```diff
@@ -0,0 +1,209 @@
+"""convert_cisagov Markdown conversion tool.
+
+Fetches the contents of the cisagov/log4j-affected-db GitHub repository
+and converts it to YAML.
```
```suggestion
Fetches the contents of the software list Markdown file in the
cisagov/log4j-affected-db GitHub repository and converts it to YAML.
```

mcdonnnj commented on src/mdyml/convert_cisagov.py:
```diff
@@ -0,0 +1,209 @@
+        if in_table:
+            logging.debug("Appending row")
+            table_rows.append(line)
+    logging.info("Done reading markdown table %d rows.", len(table_rows))
```
```suggestion
    logging.info("Done reading Markdown table with %d rows.", len(table_rows))
```

mcdonnnj commented on src/mdyml/convert_cisagov.py:
```diff
@@ -0,0 +1,209 @@
+        logging.debug(
+            "Processing row %d with %d columns: %s", row_count, len(in_row_list), row
+        )
+        if len(in_row_list) != EXPECTED_COLUMN_COUNT:
+            logging.warning(
+                "Skipping row %d with unexpected number of columns %d: %s",
+                row_count,
+                len(in_row_list),
```
```suggestion
        if (row_column_count := len(in_row_list)) != EXPECTED_COLUMN_COUNT:
            logging.warning(
                "Skipping row %d with unexpected number of columns %d: %s",
                row_count,
                row_column_count,
```

mcdonnnj reviewed (approved):
I think this second pass catches any last little bits I had in mind.

dav3r committed:
Improve log message

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

dav3r committed:
Make docstring more accurate

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

dav3r committed:
Make README more accurate

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

dav3r committed:
Move comment closer to line it applies to

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

dav3r committed:
Optimize code

Only calling len(in_row_list) once is better than calling it twice.

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

dav3r committed:
Satisfy isort pre-commit hook

dav3r committed:
Bump version from 0.0.1 to 1.0.0
